include/sparrow/memory.hpp

```diff
@@ -0,0 +1,83 @@
+#include <stdexcept>
+#include <memory>
+
+namespace sparrow {
+
+template <class T>
+class value_ptr {
+public:
+    value_ptr() = default;
+
+    explicit value_ptr(T value)
```

Comment:
I think the constructor should take a pointer instead of a value, to avoid unecessary copy when we want `value_ptr` to capture an existing pointer. A free function `make_value_ptr` can accept a value and internally allcoate a pointer and pass it to the `value_ptr` (similar to `std::make_unique_ptr`).

Reply:
I disagree, it should be able to take a pointer or a value. Both cases are useful.

Reply:
Yeah, my point was that constructing form a value could be done via a free function, to be homogeneous with the STL. But as long as both cases are supported (ivia construcrors or constructor + free function) , I'm fine.

Reply:
Not sure what would be the utility of such free function, it makes sense for shared and unique ptr because they dont do somethnig trivial, but that's not the case here.